griptape/utils/stream.py

```diff
@@ -48,14 +57,26 @@ def run(self, *args) -> Iterator[TextArtifact]:
         t = Thread(target=self._run_structure, args=args)
         t.start()
 
+        action_str = ": "
         while True:
             event = self._event_queue.get()
             if isinstance(event, FinishStructureRunEvent):
                 break
             elif isinstance(event, FinishPromptEvent):
                 yield TextArtifact(value="\n")
-            elif isinstance(event, CompletionChunkEvent):
+            elif isinstance(event, TextChunkEvent):
                 yield TextArtifact(value=event.token)
+            elif isinstance(event, ActionChunkEvent):
+                if event.tag is not None and event.name is not None and event.path is not None:
+                    yield TextArtifact(value=f"{event.name}.{event.tag}({event.path})")
+                if event.partial_input is not None:
+                    action_str += event.partial_input
+                    try:
+                        json.loads(action_str)
+                        yield TextArtifact(value=json.dumps(json.loads(action_str), indent=2))
+                        action_str = ": "
```

Comment:
Let's print the body on a new line instead of same line with `: `. More consistent with non-streaming output.